### PR TITLE
Add tests and upgrade Bazel

### PR DESCRIPTION
**Pull request by SrodriguezO**
- copy tests from lucidsoftware/rules_play_routes but use //plary-routes-compiler for the cli
- upgrade Bazel to 0.28

### CODE REVIEW

File: WORKSPACE
Comment (SrodriguezO) [inline WORKSPACE]: I'll rename this to `play_routes_compiler_cli` before merging. Holding off on getting more feedback before pushing

File: .travis.yml
Comment (borkaehw) [inline .travis.yml:12]: Could we support at least two latest versions? `0.27.2` and `0.28.1` at this moment.

File: .travis.yml
Comment (borkaehw) [inline .travis.yml:10]: Do we want to test on osx?

File: WORKSPACE
Comment (borkaehw) [inline WORKSPACE]: Why are we not using pinned json?

File: WORKSPACE
Comment (borkaehw) [inline WORKSPACE:110]: Why don't we import the artifacts in this repo and not referencing from `rules_play_routes`?

**Review by borkaehw — COMMENTED** [review]
Do we plan to remove tests in `rules_play_routes`?

File: WORKSPACE
Comment (jjudd) [reply WORKSPACE]: Thanks. `compiler_cli` doesn't seem quite specific enough.

File: WORKSPACE
Comment (jjudd) [inline WORKSPACE]: Might want to expand on this comment a bit. Java 8 needed because Scala 2.11 doesn't support Java 11 well.

File: .travis.yml
Comment (SrodriguezO) [reply .travis.yml:12]: We don't need to support multiple Bazel versions for this because the consumable is pre-built.

**Comment by SrodriguezO** [comment]
> Do we plan to remove tests in `rules_play_routes`?

No; the tests in `rules_play_routes` will be testing that the `rules_play_routes` ruleset works as expected. The tests here will be testing the the compiler CLI works as expected. 

The tests in the two repos don't need to stay in sync

File: WORKSPACE
Comment (SrodriguezO) [reply WORKSPACE]: There aren't very many dependencies, and other Bazel projects shouldn't be depending on this workspace.

File: .travis.yml
Comment (SrodriguezO) [reply .travis.yml:10]: I don't think it matters in this case, since we're testing a java executable (which is portable as long as the consumer has a valid JDK)

File: WORKSPACE
Comment (SrodriguezO) [reply WORKSPACE]: I'll update the comment

File: WORKSPACE
Comment (SrodriguezO) [reply WORKSPACE:110]: Good point; I'm actually going to remove most of the artifacts in this repo since they're not used directly

File: WORKSPACE
Comment (borkaehw) [reply WORKSPACE]: > other Bazel projects shouldn't be depending on this workspace

Agree. While if it helps speed up CI, it may still be worth it.

Comment (SrodriguezO) [reply WORKSPACE]: Good point. I'll add it.

**Review by jjudd — APPROVED** [review]

**Review by borkaehw — APPROVED** [review]
LGTM

**Pull request merged**
